src/MooLite/plugins/WhisperManager/WhisperManagerPlugin.ts

```diff
@@ -0,0 +1,138 @@
+import { ChatMessage } from "src/MooLite/core/chat/ChatMessage";
+import { ChatChannelTypeHrid } from "src/MooLite/core/chat/ChatChannelTypeHrid";
+import { ConversationMessage } from "./ConversationMessage";
+import { conversationData } from "src/MooLite/plugins/WhisperManager/DevConstants";
+import { Game } from "src/MooLite/core/Game";
+import { Conversation } from "src/MooLite/plugins/WhisperManager/Conversation";
+import { PluginConfig } from "src/MooLite/core/plugins/config/PluginConfig";
+import { PluginConfigType } from "src/MooLite/core/plugins/config/PluginConfigType";
+
+export class WhisperManagerPlugin extends MooLitePlugin {
+    name: string = "Whisper Manager";
+    key = "whisper-manager";
+    description: string = "A plugin to help manage all of your conversations!";
+
+    _messages: ChatMessage[] = [];
+    _conversations: Record<string, Conversation> = {};
+    _activeConversationName: string = "";
+
+    config: PluginConfig[] = [
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide System Messages",
+            description: "Hide system messages from the conversation list",
+            key: "hide-system-messages",
+            value: false,
+        },
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide Mod Messages",
+            description: "Hide mod messages from the conversation list",
+            key: "hide-mod-messages",
+            value: false,
+        },
+    ];
+
+    tab: MooLiteTab = {
+        icon: "💬",
+        pluginName: this.name,
+        componentName: "WhisperManagerPluginDisplay",
+        component: markRaw(WhisperManagerPluginDisplay),
+        width: "w-80",
+    };
+
+    initialize(game: Game): void {
+        super.initialize(game);
+        console.log("WhisperManagerPlugin initialized");
+    }
+
+    public get hideSystemMessages(): boolean {
+        return this.getConfig("hide-system-messages").value;
+    }
+
+    public get hideModMessages(): boolean {
+        return this.getConfig("hide-mod-messages").value;
+    }
+
+    public get activeConversation(): Conversation {
+        return this._conversations[this._activeConversationName];
+    }
+
+    public get activeConversationName(): string {
+        return this._activeConversationName;
+    }
+
+    public get messages(): ChatMessage[] {
+        return this._messages;
+    }
+
+    public get conversations(): Record<string, Conversation> {
+        let conversations: Record<string, Conversation> = this.hideSystemMessages
+            ? (({ System, ...o }) => o)(this._conversations)
+            : this._conversations;
+        conversations = this.hideModMessages ? (({ Mods, ...o }) => o)(conversations) : conversations;
+
+        return conversations;
+    }
+
+    public get game(): Game {
+        return this._game;
+    }
+
+    public populateDevConversations(): void {
```

Comment:
Also remove/move to test

Reply:
pending [this question](https://github.com/Ishadijcks/MooLite/pull/96#discussion_r1243000427).